File: app/src/main/java/com/qiaoqiao/core/camera/vision/ui/viewholder/LogoViewHolder.java
package com.qiaoqiao.core.camera.vision.ui.viewholder;

import android.support.annotation.NonNull;
import android.view.View;

import com.bumptech.glide.Glide;
import com.qiaoqiao.core.camera.vision.model.VisionEntity;
import com.qiaoqiao.databinding.LogoViewBinding;

import java.util.List;

public final class LogoViewHolder extends AbstractVisionViewHolder {
	private final @NonNull LogoViewBinding mItemVisionLogoBinding;

	public LogoViewHolder(@NonNull LogoViewBinding binding, @NonNull List<VisionEntity> entities) {
		super(binding, entities);
		mItemVisionLogoBinding = binding;
	}

	@Override
	public void onBindViewHolder() {
		VisionEntity entity = mEntities.get(getAdapterPosition());
		mItemVisionLogoBinding.visionTv.setText(entity.getDescription()
		                                             .getDescriptionText());
		mItemVisionLogoBinding.setVisionEntity(entity);
		mItemVisionLogoBinding.setViewholder(this);
		mItemVisionLogoBinding.visionIv.setActivated(entity.isActivated());
		mItemVisionLogoBinding.visionTvFl.setActivated(entity.isActivated());
		loadImage(itemView.getContext(), entity, mItemVisionLogoBinding.visionIv);
	}

	@Override
	public void onViewRecycled() {
		Glide.clear(mItemVisionLogoBinding.visionIv);
	}

	@Override
	protected View getTransitionView() {
		return mItemVisionLogoBinding.visionIv;
	}
}

File: app/src/main/java/com/qiaoqiao/core/camera/vision/ui/viewholder/LogoViewHolder.kt
package com.qiaoqiao.core.camera.vision.ui.viewholder

import android.view.View
import com.qiaoqiao.app.GlideApp
import com.qiaoqiao.core.camera.vision.model.VisionEntity
import com.qiaoqiao.databinding.LogoViewBinding

internal class LogoViewHolder(private val itemVisionLogoBinding: LogoViewBinding, entities: List<VisionEntity>) : AbstractVisionViewHolder(itemVisionLogoBinding.root, entities) {
    override fun onBindViewHolder() {
        val entity = entities[adapterPosition]
        itemVisionLogoBinding.visionTv.text = entity.description
                .descriptionText
        itemVisionLogoBinding.visionEntity = entity
        itemVisionLogoBinding.viewholder = this
        itemVisionLogoBinding.visionIv.isActivated = entity.isActivated
        itemVisionLogoBinding.visionTvFl.isActivated = entity.isActivated
        loadImage(itemView.context, entity, itemVisionLogoBinding.visionIv)
    }

    override fun onViewRecycled() {
        GlideApp.with(itemVisionLogoBinding.visionIv).clear(itemVisionLogoBinding.visionIv)
    }

    override fun getTransitionView(): View {
        return itemVisionLogoBinding.visionIv
    }

    override fun getBinding() = itemVisionLogoBinding
}